src/teachy_tv.c

```diff
@@ -0,0 +1,1622 @@
+#include "main.h"
+#include "global.h"
+#include "task.h"
+#include "menu.h"
+#include "palette.h"
+#include "sprite.h"
+#include "menu_helpers.h"
+#include "new_menu_helpers.h"
+#include "sound.h"
+#include "malloc.h"
+#include "scanline_effect.h"
+#include "bg.h"
+#include "gpu_regs.h"
+#include "decompress.h"
+#include "window.h"
+#include "list_menu.h"
+#include "item_menu.h"
+#include "item.h"
+#include "menu_indicators.h"
+#include "field_map_obj.h"
+#include "random.h"
+#include "text.h"
+#include "constants/songs.h"
+#include "constants/items.h"
+#include "event_data.h"
+#include "load_save.h"
+#include "battle_transition.h"
+#include "battle_2.h"
+#include "battle.h"
+#include "global.fieldmap.h"
+#include "teachy_tv.h"
+
+struct TeachyTvCtrlBlk
+{
+    void (*callback)();
+    u8 mode;
+    u8 optionChosen;
+    u16 scrollOffset;
+    u16 selectedRow;
+    u16 filler;
+};
+
+struct TeachyTvBuf
+{
+    u32 state;
+    u8 buffer1[0x1000];
+    u8 buffer2[0x1000];
+    u8 buffer3[0x1000];
+    u8 buffer4[0x1000];
+    u8 var_4004;
+    u8 var_4005;
+    u8 var_4006;
+    u8 var_4007;
+};
+
+EWRAM_DATA struct TeachyTvCtrlBlk gTeachyTV_StaticResources = {0};
+extern struct TeachyTvBuf * gUnknown_203F450;
+extern const struct ScrollIndicatorArrowPairTemplate gUnknown_8479380;
+extern const u8 gUnknown_841B83D;
+extern const struct BgTemplate gUnknown_84792E0;
+extern const struct SpritePalette gUnknown_83A5348;
+extern void (**gUnknown_8479548)(u8);
+extern const u8 * const gUnknown_8479560;
+extern const u8 * const gUnknown_8479578;
+extern const u8 gUnknown_8E86240;
+extern const u8 gUnknown_8E86BE8;
+extern const u8 gUnknown_8E86D6C;
+extern const u8 gUnknown_8E86F98;
+extern const struct ListMenuTemplate gUnknown_8479368;
+extern const struct ListMenuItem gUnknown_8479340;
+extern const struct WindowTemplate gUnknown_84792F0;
+extern const u8 gUnknown_8479590;
+extern const u8 gUnknown_8479390;
+extern const struct SubspriteTable gUnknown_84795B8;
+extern const struct SpriteTemplate * const gUnknown_83A0010;
+extern const u8 gUnknown_84795C8;
+extern const struct MapData Route1_Layout;
+
+extern void sub_812B1E0(u16);
```

Comment:
Find out where this and the following four functions are located in ROM, and define them in their respective headers; make sure those headers are included; and get these declarations the f out of here